atlassian_jwt_auth/frameworks/common/backend.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+from atlassian_jwt_auth import HTTPSPublicKeyRetriever, JWTAuthVerifier
+
+from jwt.compat import text_type
+from .utils import SettingsDict
+
+
+class Backend():
+    """Abstract class representing a web framework backend
+
+    Backends allow specific implementation details of web frameworks to be
+    abstracted away from the underlying logic of ASAP.
+    """
+    __metaclass__ = ABCMeta
+
+    default_headers_401 = {'WWW-Authenticate': 'Bearer'}
+    default_settings = {
+        # The class to be instantiated to retrieve public keys
+        'ASAP_KEY_RETRIEVER_CLASS': HTTPSPublicKeyRetriever,
+
+        # The repository URL where the key retriever can fetch public keys
+        'ASAP_PUBLICKEY_REPOSITORY': None,
+
+        # Whether or not ASAP authentication is required
+        # This is primarily useful when phasing in ASAP authentication
+        'ASAP_REQUIRED': True,
+
+        # The valid audience valuw expected when authenticating tokens
+        'ASAP_VALID_AUDIENCE': None,
+
+        # The amount of leeway to apply when evaluating token expiration
+        # timestamps
+        'ASAP_VALID_LEEWAY': 0,
+
+        # A list of valid token issuers allowed to authenticate
+        # (this can be overridden at the decorator level)
+        'ASAP_VALID_ISSUERS': None,
+
+        # Enforce that the ASAP subject must match the issuer
+        'ASAP_SUBJECT_SHOULD_MATCH_ISSUER': False
```

Comment:
Shouldn't we default this to `None`?

Reply:
No, I think that would be more ambiguous. It's clear to the consumer here that if you don't override the config, the subject does not have to match the issuer. If we set it to `None`, it wouldn't be clear... is it on or is it off?

Reply:
My preference is for to take on the value that we use as the default in `JWTAuthVerifier`.

Reply:
What are your thoughts on making this the default? It seems silly to, by default, require ASAP tokens to include a duplicative subject key for no other reason than to match the issuer.

Reply:
If we are going to change the default behaviour, let's do it in another pr and not this one :-).